### PR TITLE
To Do list: list structure

### PR DESCRIPTION
**Pull request by zmekonnen251**
### General requirements

- [x] There are no linter errors.
- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### JavaScript requirements

- [x] Followed list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### Project requirements
- [x] Set up a new project with webpack .
- [x] Created an **index.html** file and wrote HTML markup here. Created an empty **To Do List** placeholder (`<ul>` element). The **index.html** file must be set as a template using the *HTML Webpack Plugin*.
- [x] Created an **index.js** file and set an array of some simple *to do tasks* (array of objects). Each task object contain three keys: 
  1. description [`string`].
  2. completed [`bool`].
  3. index: [`number`].
- [x] Wrote a function to iterate over the tasks array and populate an HTML list item element for each task.
- [x] On page load render the dynamically created list of tasks in the dedicated placeholder. The list appears in order of the `index` values for each task.
- [x] Created a **style.css** and set rules for the **To Do List**. CSS must be loaded by *Webpack Style/CSS Loader*. The list is a clone of the part of the minimalist project captured in the video below.

[Capstone intro](https://www.youtube.com/watch?v=AcUd-_Yjjqg)
- [x] All your source files (**index.html**, **index.js** and **style.css**) must be located in */src* directory and your distribution files will be generated by webpack and served by *webpack dev server* from */dist* folder.

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "to-do-list",
+  "version": "1.0.0",
+  "description": "![](https://img.shields.io/badge/Microverse-blueviolet)",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack"
```
Comment (elyor-doniyorov) [inline package.json]: - [x] I recommend adding Webpack `start` and `watch` scripts to make `npm start` possible to start the app. You can add the following lines of code in your scripts : `"watch": "webpack --watch",`
    `"start": "webpack serve --open",`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "eslint-plugin-import": "^2.25.4",
+    "hint": "^6.1.9",
+    "html-webpack-plugin": "^5.5.0",
+    "style-loader": "^3.3.1",
+    "stylelint": "^13.13.1",
+    "stylelint-config-standard": "^21.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.21.0",
+    "webpack": "^5.69.0",
+    "webpack-cli": "^4.9.2"
```
Comment (elyor-doniyorov) [inline package.json]: - [x] Make sure the Webpack devserver is setup and running with no errors. To implement this you can run the following command : `npm install --save-dev webpack webpack-dev-server`

**Review by elyor-doniyorov — CHANGES_REQUESTED** [review]
Hi @zmekonnen251, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Zelalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**